bindings/go/oci/repository_test.go

```diff
@@ -893,3 +1077,330 @@ func setupLegacyComponentVersion(t *testing.T, store *ocictf.Store, ctx context.
+						runtime.NewVersionedType(v2.LocalBlobAccessType, v2.LocalBlobAccessTypeVersion),
+						digest.FromString("platform specific content").String(),
+					)),
+				},
+			},
+			content: []byte("platform specific content"),
+			identity: map[string]string{
+				"name":         "platform-source",
+				"version":      "1.0.0",
+				"architecture": "amd64",
+				"os":           "linux",
+			},
+			setupComponent: true,
+			checkContent: func(t *testing.T, original []byte, actual []byte) {
+				assert.Equal(t, string(original), string(actual))
+			},
+		},
+		{
+			name: "source with invalid identity",
+			source: &descriptor.Source{
+				ElementMeta: descriptor.ElementMeta{
+					ObjectMeta: descriptor.ObjectMeta{
+						Name:    "platform-source",
+						Version: "1.0.0",
+					},
+					ExtraIdentity: map[string]string{
+						"architecture": "amd64",
+						"os":           "linux",
+					},
+				},
+				Type: "test-type",
+				Access: &runtime.Raw{
+					Type: runtime.NewVersionedType(v2.LocalBlobAccessType, v2.LocalBlobAccessTypeVersion),
+					Data: []byte(fmt.Sprintf(
+						`{"type":"%s","localReference":"%s","mediaType":"application/octet-stream"}`,
+						runtime.NewVersionedType(v2.LocalBlobAccessType, v2.LocalBlobAccessTypeVersion),
+						digest.FromString("platform specific content").String(),
+					)),
+				},
+			},
+			identity: map[string]string{
+				"name":    "test-source",
+				"version": "1.0.0",
+			},
+			expectError:    true,
+			errorContains:  "found 0 candidates while looking for source \"name=test-source,version=1.0.0\", but expected exactly one",
+			setupComponent: true,
+		},
+		{
+			name: "oci layout source",
+			source: &descriptor.Source{
+				ElementMeta: descriptor.ElementMeta{
+					ObjectMeta: descriptor.ObjectMeta{
+						Name:    "oci-layout-source",
+						Version: "1.0.0",
+					},
+				},
+				Type: "test-type",
+				Access: &runtime.Raw{
+					Type: runtime.NewVersionedType(v2.LocalBlobAccessType, v2.LocalBlobAccessTypeVersion),
+					Data: []byte(fmt.Sprintf(
+						`{"type":"%s","localReference":"%s","mediaType":"%s"}`,
+						runtime.NewVersionedType(v2.LocalBlobAccessType, v2.LocalBlobAccessTypeVersion),
+						digest.FromString("oci layout content").String(),
+						layout.MediaTypeOCIImageLayoutV1+"+tar+gzip",
+					)),
+				},
+			},
+			content: func() []byte {
+				// Create a buffer to hold the OCI layout
+				buf := bytes.NewBuffer(nil)
+				layout := tar.NewOCILayoutWriter(buf)
+
+				// Create a descriptor for our content
+				content := []byte("oci layout content")
+				desc := ociImageSpecV1.Descriptor{
+					MediaType: ociImageSpecV1.MediaTypeImageLayer,
+					Digest:    digest.FromBytes(content),
+					Size:      int64(len(content)),
+				}
+
+				// Push the content
+				if err := layout.Push(t.Context(), desc, bytes.NewReader(content)); err != nil {
+					panic(fmt.Sprintf("failed to push content: %v", err))
```

Comment:
You could just pass in r to this and call r.NoError? Totally up to you, a panic here IS weird... :D

Reply:
holy moly youre right.